### DISCUSSION TITLE
Automatic deploy doesnt load environment variables.

### DISCUSSION THREAD

**Discussion by iNetJoJo** [vercel/community#4844] [Help] [answered]
### Page to Investigate

deployments

### Steps to Reproduce

1. create new project nextjs or vite
2. add envoronemnt variable redeploy (env good)
3. commit new change (redeploy is faster but env are empty in the app)
4. force redeploy (env are now ok)

**Comment by amyegan** [comment]
I wasn't able to reproduce this issue. Did you [declare the environment variable](https://vercel.com/docs/projects/environment-variables#declare-an-environment-variable) in your project settings on Vercel, or did you add them a different way?

**Reply by amyegan** [reply]
If you're using the environment variable in client side code, you may also need to include a prefix to indicate that it is safe for front end use. For example, Next.js uses the prefix [`NEXT_PUBLIC_`](https://nextjs.org/docs/pages/building-your-application/configuring/environment-variables#bundling-environment-variables-for-the-browser) and Vite uses the prefix [`VITE_`](https://vitejs.dev/guide/env-and-mode#env-files) make sure sensitive values are not exposed to the browser.

**Reply by iNetJoJo** [reply]
so you are saying if i have NEXT_PUBLIC or VITE_ in my envornment variables. caching build steps will know do detect this?

**Reply by iNetJoJo** [reply]
@amyegan i also had an issue today where automatic deploy deployed my frontend with staging environment variables. i had to run redeploy with cache build unchecked, and it deployed fine.

**Reply by amyegan** [reply]
Do you happen to be using Turborepo or Nx?

**Reply by iNetJoJo** [reply]
Turborepo

**Reply by amyegan** [reply, marked as answer]
It could be a remote cache problem. The Turborepo docs have a [Configuring Cache Outputs](https://turbo.build/repo/docs/core-concepts/caching/what-to-cache#configuring-cache-outputs) page that can help if caching is the source of the problem.

If that's not it, I could use a [minimal reproducible example](https://vercel.com/support/articles/creating-a-minimal-reproducible-example) to dig into the problem further.

**Comment by ststevanovic** [comment]
In my case, both redeploy and autodeploy fail to register ENV variable...
Keep getting a message:
Uncaught (in promise) Error: Missing required parameters: sitekey
    at Ab.<anonymous> (recaptcha__en.js:335:315)
I checked the name for the process.env variable, not it.
I use google reCAPTCHA, so i registered the vercel domain i got to google reCAPTCHA panel...
the reCAPTCHA sitekey works on local

Any help would be appreciated 
thanks in advance
